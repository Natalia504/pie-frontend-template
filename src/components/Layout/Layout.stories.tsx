import React from "react";
import { storiesOf } from "@storybook/react";
import { withKnobs, text } from "@storybook/addon-knobs";
import { withA11y } from "@storybook/addon-a11y";
import { Typography } from "@material-ui/core";

import { Layout } from "./Layout";

storiesOf("Layout", module)
  .addDecorator(withA11y)
  .addDecorator(withKnobs)
  .add("Kitchen Sink", () => {
    const header = text("Header", "Test Header");
    return (
      <Layout {...{ header }}>
        <Typography variant="body2" color="textSecondary">
          Layout Story
        </Typography>
      </Layout>
    );
  });
