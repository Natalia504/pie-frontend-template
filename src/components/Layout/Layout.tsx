import React from "react";
import { Typography } from "@material-ui/core";

export interface LayoutProps {
  header?: string;
}

export const Layout: React.FC<LayoutProps> = ({
  header = "App Header",
  children
}) => (
  <div>
    <Typography>{header}</Typography>
    {children}
  </div>
);
