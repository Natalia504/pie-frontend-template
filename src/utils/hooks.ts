import { useEffect } from "react";

export const useMount = (
  mountFunction: () => void,
  cleanupFunction?: () => void
): void => {
  useEffect(() => {
    mountFunction();
    return cleanupFunction && cleanupFunction();
  }, []); /* eslint-disable-line react-hooks/exhaustive-deps */
};
