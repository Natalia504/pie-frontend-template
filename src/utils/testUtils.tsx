import React from "react";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import { ThemeProvider, createMuiTheme } from "@material-ui/core";
import { pieLightTheme } from "@pie/components";

const mockState = {
  counter: 0
};

export const mockStore = configureMockStore()(mockState);

export const renderWithProviders = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult =>
  render(
    <Provider store={mockStore}>
      <ThemeProvider theme={createMuiTheme(pieLightTheme)}>{ui}</ThemeProvider>
    </Provider>
  );
