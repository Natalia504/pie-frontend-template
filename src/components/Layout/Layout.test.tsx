import React from "react";
import { RenderResult } from "@testing-library/react";

import { Layout } from "./Layout";
import { renderWithProviders } from "../../utils/testUtils";

describe("Layout tests", () => {
  const setup = (): RenderResult =>
    renderWithProviders(
      <Layout>
        <div>I'm a child</div>
      </Layout>
    );
  it("renders", () => {
    const { baseElement } = setup();
    expect(baseElement).toMatchSnapshot();
  });
});
