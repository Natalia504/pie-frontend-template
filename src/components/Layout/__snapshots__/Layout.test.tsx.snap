// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout tests renders 1`] = `
<body>
  <div>
    <div>
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        App Header
      </p>
      <div>
        I'm a child
      </div>
    </div>
  </div>
</body>
`;
